Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-17.ca-central-1.compute.internal
Sha: 23a46b8bd4fb8b006602c2692f2a55597ce9619e
Build: success
Install warning(s): 
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.11 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.11: wanted {os:darwin,arch:any} (current: {os:linux,arch:x64})

Lint error(s): 
eclwatch/stub.js: line 27, col 5, const is available in ES6 (use esversion: 6) or Mozilla JS extensions (use moz).
1 error
res:
> eclwatch@1.0.0 test /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> run-s jslint tslint
> eclwatch@1.0.0 jslint /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
npm ERR! code ELIFECYCLE
res:
> eclwatch@1.0.0 test /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> run-s jslint tslint
> eclwatch@1.0.0 jslint /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
npm ERR! errno 2
res:
> eclwatch@1.0.0 test /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> run-s jslint tslint
> eclwatch@1.0.0 jslint /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
npm ERR! eclwatch@1.0.0 jslint: `jshint --config ./.jshintrc ./eclwatch`
res:
> eclwatch@1.0.0 test /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> run-s jslint tslint
> eclwatch@1.0.0 jslint /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
npm ERR! Exit status 2
res:
> eclwatch@1.0.0 test /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> run-s jslint tslint
> eclwatch@1.0.0 jslint /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
npm ERR! 
res:
> eclwatch@1.0.0 test /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> run-s jslint tslint
> eclwatch@1.0.0 jslint /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
npm ERR! Failed at the eclwatch@1.0.0 jslint script.
res:
> eclwatch@1.0.0 test /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> run-s jslint tslint
> eclwatch@1.0.0 jslint /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
res:
> eclwatch@1.0.0 test /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> run-s jslint tslint
> eclwatch@1.0.0 jslint /home/centos/smoketest/PR-13371/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
npm ERR! A complete log of this run can be found in:\n ... ( comment is too long, 192 lines, 8633 bytes, truncated) ... 
[Build error:](https://gist.github.com/a92b592e1b43581b2b672b637cace0e6)
Build log [Build log](https://gist.github.com/HPCCSmoketest/a92b592e1b43581b2b672b637cace0e6#file-result-20-02-19-07-20-51-log)

Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   36 sec (00:00:36) |  767 sec (00:12:47) |  122 sec (00:02:02) |   23 sec (00:00:23) |   17 sec (00:00:17) |    0 sec (00:00:00) |   16 sec (00:00:16) |  981 sec (00:16:21) |

```

--- a/esp/src/eclwatch/dojoConfig.js
+++ b/esp/src/eclwatch/dojoConfig.js
@@ -1,7 +1,10 @@
+var fs = require("fs");
+
 function getConfig(env) {
     // dojoRoot is defined if we're running in node (i.e. building)
     var dojoRoot = env.dojoRoot;
     var baseUrl = dojoRoot ? "." : "/esp/files";
+    var hpccBaseUrl = fs.existsSync(baseUrl + "/node_modules/@hpcc-js") ? baseUrl + "/node_modules/@hpcc-js" : baseUrl + "/../../../hpcc-js/packages";
 
     return {
         baseUrl: baseUrl,
@@ -20,29 +23,29 @@ function getConfig(env) {
             "templates": baseUrl + "/eclwatch/templates",
             "ecl": baseUrl + "/eclwatch/ecl",
             "css": baseUrl + "/loader/css",
-            "@hpcc-js/api": baseUrl + "/node_modules/@hpcc-js/api/dist/index",
-            "@hpcc-js/chart": baseUrl + "/node_modules/@hpcc-js/chart/dist/index",
-            "@hpcc-js/common": baseUrl + "/node_modules/@hpcc-js/common/dist/index",
-            "@hpcc-js/comms": baseUrl + "/node_modules/@hpcc-js/comms/dist/index",
-            "@hpcc-js/composite": baseUrl + "/node_modules/@hpcc-js/composite/dist/index",
-            "@hpcc-js/dgrid": baseUrl + "/node_modules/@hpcc-js/dgrid/dist/index",
-            "@hpcc-js/dgrid-shim": baseUrl + "/node_modules/@hpcc-js/dgrid-shim/dist/index",
-            "@hpcc-js/eclwatch": baseUrl + "/node_modules/@hpcc-js/eclwatch/dist/index",
-            "@hpcc-js/form": baseUrl + "/node_modules/@hpcc-js/form/dist/index",
-            "@hpcc-js/graph": baseUrl + "/node_modules/@hpcc-js/graph/dist/index",
-            "@hpcc-js/layout": baseUrl + "/node_modules/@hpcc-js/layout/dist/index",
-            "@hpcc-js/html": baseUrl + "/node_modules/@hpcc-js/html/dist/index",
-            "@hpcc-js/map": baseUrl + "/node_modules/@hpcc-js/map/dist/index",
-            "@hpcc-js/other": baseUrl + "/node_modules/@hpcc-js/other/dist/index",
-            "@hpcc-js/react": baseUrl + "/node_modules/@hpcc-js/react/dist/index",
-            "@hpcc-js/timeline": baseUrl + "/node_modules/@hpcc-js/timeline/dist/index",
-            "@hpcc-js/tree": baseUrl + "/node_modules/@hpcc-js/tree/dist/index",
-            "@hpcc-js/util": baseUrl + "/node_modules/@hpcc-js/util/dist/index",
-            "@hpcc-js/TopoJSON": dojoRoot ? "/esp/files/dist/TopoJSON" : baseUrl + "/node_modules/@hpcc-js/map/TopoJSON",
+            "@hpcc-js/api": hpccBaseUrl + "/api/dist/index",
+            "@hpcc-js/chart": hpccBaseUrl + "/chart/dist/index",
+            "@hpcc-js/common": hpccBaseUrl + "/common/dist/index",
+            "@hpcc-js/comms": hpccBaseUrl + "/comms/dist/index",
+            "@hpcc-js/composite": hpccBaseUrl + "/composite/dist/index",
+            "@hpcc-js/dgrid": hpccBaseUrl + "/dgrid/dist/index",
+            "@hpcc-js/dgrid-shim": hpccBaseUrl + "/dgrid-shim/dist/index",
+            "@hpcc-js/eclwatch": hpccBaseUrl + "/eclwatch/dist/index",
+            "@hpcc-js/form": hpccBaseUrl + "/form/dist/index",
+            "@hpcc-js/graph": hpccBaseUrl + "/graph/dist/index",
+            "@hpcc-js/layout": hpccBaseUrl + "/layout/dist/index",
+            "@hpcc-js/html": hpccBaseUrl + "/html/dist/index",
+            "@hpcc-js/map": hpccBaseUrl + "/map/dist/index",
+            "@hpcc-js/other": hpccBaseUrl + "/other/dist/index",
+            "@hpcc-js/react": hpccBaseUrl + "/react/dist/index",
+            "@hpcc-js/timeline": hpccBaseUrl + "/timeline/dist/index",
+            "@hpcc-js/tree": hpccBaseUrl + "/tree/dist/index",
+            "@hpcc-js/util": hpccBaseUrl + "/util/dist/index",
+            "@hpcc-js/TopoJSON": dojoRoot ? "/esp/files/dist/TopoJSON" : hpccBaseUrl + "/map/TopoJSON",
             "clipboard": baseUrl + "/node_modules/clipboard/dist/clipboard",
             "codemirror": baseUrl + "/node_modules/codemirror",
             "crossfilter": baseUrl + "/node_modules/crossfilter2/crossfilter.min",
-            "font-awesome": baseUrl + "/node_modules/@hpcc-js/common/font-awesome",
+            "font-awesome": hpccBaseUrl + "/common/font-awesome",
             "tslib": baseUrl + "/node_modules/tslib/tslib"
         },
         packages: [

--- a/esp/src/package.json
+++ b/esp/src/package.json
@@ -27,8 +27,7 @@
     "dev-start-ws": "ws",
     "dev-start": "run-p bundle-watch dev-start-ws",
     "dev-start-verbose": "ws --verbose.include request response",
-    "dev-link": "rimraf ./node_modules/@hpcc-js && mklink /J .\\node_modules\\@hpcc-js ..\\..\\..\\hpcc-js\\packages",
-    "dev-unlink": "rimraf ./node_modules/@hpcc-js && npm install",
+    "rm-hpcc": "rimraf ./node_modules/@hpcc-js",
     "test": "run-s lint"
   },
   "main": "src/stub.js",

--- a/esp/src/tsconfig.json
+++ b/esp/src/tsconfig.json
@@ -1,5 +1,6 @@
 {
     "compilerOptions": {
+        "baseUrl": ".",
         "outDir": "./lib",
         "target": "es5",
         "module": "amd",
@@ -21,7 +22,13 @@
             "es2015"
         ],
         "typeRoots": [],
-        "types": []
+        "types": [],
+        "paths": {
+            "@hpcc-js/*": [
+                "./node_modules/@hpcc-js/*",
+                "../../../hpcc-js/packages/*"
+            ]
+        }
     },
     "include": [
         "src/**/*"

